`test_custom_interpreter` fails

### What pipdeptree version are you using?

2.17.0

### Are you running pipdeptree in a virtual environment?

- [ ] Yes

### Describe the problem

With version `2.17.0` the `test_custom_interpreter` test started to fail:
```
============================= test session starts ==============================
platform sunos5 -- Python 3.9.19, pytest-8.1.1, pluggy-1.4.0 -- $(BUILD_DIR)/.tox/py39/bin/python
cachedir: .tox/py39/.pytest_cache
rootdir: $(BUILD_DIR)
configfile: pyproject.toml
plugins: mock-3.14.0, cov-5.0.0
collecting ... collected 2 items

tests/test_non_host.py::test_custom_interpreter[True] FAILED             [ 50%]
tests/test_non_host.py::test_custom_interpreter[False] FAILED            [100%]

=================================== FAILURES ===================================
________________________ test_custom_interpreter[True] _________________________

tmp_path = PosixPath('/tmp/pytest-of-marcel/pytest-137/test_custom_interpreter_True_0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fffacde7130>
capfd = <_pytest.capture.CaptureFixture object at 0x7fffacde7310>
args_joined = True

    @pytest.mark.parametrize("args_joined", [True, False])
    def test_custom_interpreter(
        tmp_path: Path,
        monkeypatch: pytest.MonkeyPatch,
        capfd: pytest.CaptureFixture[str],
        args_joined: bool,
    ) -> None:
        result = virtualenv.cli_run([str(tmp_path / "venv"), "--activators", ""])
        cmd = [sys.executable]
        monkeypatch.chdir(tmp_path)
        py = str(result.creator.exe.relative_to(tmp_path))
        cmd += [f"--python={result.creator.exe}"] if args_joined else ["--python", py]
        monkeypatch.setattr(sys, "argv", cmd)
        main()
        out, _ = capfd.readouterr()
        found = {i.split("==")[0] for i in out.splitlines()}
        implementation = python_implementation()
        if implementation == "CPython":
            expected = {"pip", "setuptools", "wheel"}
        elif implementation == "PyPy":
            expected = {"cffi", "greenlet", "pip", "readline", "setuptools", "wheel"}  # pragma: no cover
        else:
            raise ValueError(implementation)
        if sys.version_info >= (3, 12):
            expected -= {"setuptools", "wheel"}
>       assert found == expected, out
E       AssertionError: pipdeptree==2.17.0
E         ├── packaging [required: >=23.1, installed: 24.0]
E         └── pip [required: >=23.1.2, installed: 24.0]
E         setuptools==69.1.1
E         wheel==0.42.0
E
E       assert {'pipdeptree', '├── packaging [required: >=23.1, installed: 24.0]', 'wheel', '└── pip [required: >=23.1.2, installed: 24.0]', 'setuptools'} == {'pip', 'setuptools', 'wheel'}
E
E         Extra items in the left set:
E         '├── packaging [required: >=23.1, installed: 24.0]'
E         '└── pip [required: >=23.1.2, installed: 24.0]'
E         'pipdeptree'
E         Extra items in the right set:
E         'pip'
E
E         Full diff:
E           {
E         -     'pip',
E         +     'pipdeptree',
E               'setuptools',
E               'wheel',
E         +     '└── pip [required: >=23.1.2, installed: 24.0]',
E         +     '├── packaging [required: >=23.1, installed: 24.0]',
E           }

$(BUILD_DIR)/tests/test_non_host.py:41: AssertionError
________________________ test_custom_interpreter[False] ________________________

tmp_path = PosixPath('/tmp/pytest-of-marcel/pytest-137/test_custom_interpreter_False_0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fffacc09d00>
capfd = <_pytest.capture.CaptureFixture object at 0x7fffacc097f0>
args_joined = False

    @pytest.mark.parametrize("args_joined", [True, False])
    def test_custom_interpreter(
        tmp_path: Path,
        monkeypatch: pytest.MonkeyPatch,
        capfd: pytest.CaptureFixture[str],
        args_joined: bool,
    ) -> None:
        result = virtualenv.cli_run([str(tmp_path / "venv"), "--activators", ""])
        cmd = [sys.executable]
        monkeypatch.chdir(tmp_path)
        py = str(result.creator.exe.relative_to(tmp_path))
        cmd += [f"--python={result.creator.exe}"] if args_joined else ["--python", py]
        monkeypatch.setattr(sys, "argv", cmd)
        main()
        out, _ = capfd.readouterr()
        found = {i.split("==")[0] for i in out.splitlines()}
        implementation = python_implementation()
        if implementation == "CPython":
            expected = {"pip", "setuptools", "wheel"}
        elif implementation == "PyPy":
            expected = {"cffi", "greenlet", "pip", "readline", "setuptools", "wheel"}  # pragma: no cover
        else:
            raise ValueError(implementation)
        if sys.version_info >= (3, 12):
            expected -= {"setuptools", "wheel"}
>       assert found == expected, out
E       AssertionError: pipdeptree==2.17.0
E         ├── packaging [required: >=23.1, installed: 24.0]
E         └── pip [required: >=23.1.2, installed: 24.0]
E         setuptools==69.1.1
E         wheel==0.42.0
E
E       assert {'pipdeptree', '├── packaging [required: >=23.1, installed: 24.0]', 'wheel', '└── pip [required: >=23.1.2, installed: 24.0]', 'setuptools'} == {'pip', 'setuptools', 'wheel'}
E
E         Extra items in the left set:
E         '├── packaging [required: >=23.1, installed: 24.0]'
E         '└── pip [required: >=23.1.2, installed: 24.0]'
E         'pipdeptree'
E         Extra items in the right set:
E         'pip'
E
E         Full diff:
E           {
E         -     'pip',
E         +     'pipdeptree',
E               'setuptools',
E               'wheel',
E         +     '└── pip [required: >=23.1.2, installed: 24.0]',
E         +     '├── packaging [required: >=23.1, installed: 24.0]',
E           }

$(BUILD_DIR)/tests/test_non_host.py:41: AssertionError
```
This is regression.  When I try the same with version `2.16.2` the test pass as expected.

--- a/tests/test_non_host.py
+++ b/tests/test_non_host.py
@@ -103,6 +103,6 @@ def test_custom_interpreter_ensure_pythonpath_envar_is_honored(
         expected = {"foo", "cffi", "greenlet", "pip", "readline", "setuptools", "wheel"}
     else:  # pragma: no cover
         raise ValueError(implementation)
-    if sys.version_info >= (3, 12):  # pragma: <3.12 no cover
+    if sys.version_info >= (3, 12):  # pragma: >=3.12 cover
         expected -= {"setuptools", "wheel"}
     assert found == expected, out

--- a/tests/test_non_host.py
+++ b/tests/test_non_host.py
@@ -99,10 +99,10 @@ def test_custom_interpreter_ensure_pythonpath_envar_is_honored(
     implementation = python_implementation()
     if implementation == "CPython":
         expected = {"foo", "pip", "setuptools", "wheel"}
-    elif implementation == "PyPy":
-        expected = {"foo", "cffi", "greenlet", "pip", "readline", "setuptools", "wheel"}  # pragma: no cover
-    else:
+    elif implementation == "PyPy":  # pragma: cpython no cover
+        expected = {"foo", "cffi", "greenlet", "pip", "readline", "setuptools", "wheel"}
+    else:  # pragma: no cover
         raise ValueError(implementation)
-    if sys.version_info >= (3, 12):
+    if sys.version_info >= (3, 12):  # pragma: <3.12 no cover
         expected -= {"setuptools", "wheel"}
     assert found == expected, out

--- a/tests/test_non_host.py
+++ b/tests/test_non_host.py
@@ -23,11 +23,14 @@ def test_custom_interpreter(
     capfd: pytest.CaptureFixture[str],
     args_joined: bool,
 ) -> None:
-    result = virtualenv.cli_run([str(tmp_path / "venv"), "--activators", ""])
-    cmd = [sys.executable]
+    # Delete $PYTHONPATH so that it cannot be passed to the custom interpreter process (since we don't know what
+    # distribution metadata to expect when it's used).
+    monkeypatch.delenv("PYTHONPATH", False)
+
     monkeypatch.chdir(tmp_path)
+    result = virtualenv.cli_run([str(tmp_path / "venv"), "--activators", ""])
     py = str(result.creator.exe.relative_to(tmp_path))
-    cmd += [f"--python={result.creator.exe}"] if args_joined else ["--python", py]
+    cmd = ["", f"--python={result.creator.exe}"] if args_joined else ["", "--python", py]
     mocker.patch("pipdeptree._discovery.sys.argv", cmd)
     main()
     out, _ = capfd.readouterr()
@@ -68,4 +71,38 @@ def test_custom_interpreter_with_local_only(
         raise ValueError(implementation)  # pragma: no cover
     if sys.version_info >= (3, 12):
         expected -= {"setuptools", "wheel"}  # pragma: no cover
+    assert found == expected, out
+
+
+def test_custom_interpreter_ensure_pythonpath_envar_is_honored(
+    tmp_path: Path,
+    mocker: MockerFixture,
+    monkeypatch: pytest.MonkeyPatch,
+    capfd: pytest.CaptureFixture[str],
+) -> None:
+    # ensures that we honor $PYTHONPATH when passing it to the custom interpreter process
+    venv_path = str(tmp_path / "venv")
+    result = virtualenv.cli_run([venv_path, "--activators", ""])
+
+    another_path = tmp_path / "another-path"
+    fake_dist = another_path / "foo-1.2.3.dist-info"
+    fake_dist.mkdir(parents=True)
+    fake_metadata = fake_dist / "METADATA"
+    with fake_metadata.open("w") as f:
+        f.write("Metadata-Version: 2.3\n" "Name: foo\n" "Version: 1.2.3\n")
+    cmd = ["", f"--python={result.creator.exe}"]
+    mocker.patch("pipdeptree._discovery.sys.argv", cmd)
+    monkeypatch.setenv("PYTHONPATH", str(another_path))
+    main()
+    out, _ = capfd.readouterr()
+    found = {i.split("==")[0] for i in out.splitlines()}
+    implementation = python_implementation()
+    if implementation == "CPython":
+        expected = {"foo", "pip", "setuptools", "wheel"}
+    elif implementation == "PyPy":
+        expected = {"foo", "cffi", "greenlet", "pip", "readline", "setuptools", "wheel"}  # pragma: no cover
+    else:
+        raise ValueError(implementation)
+    if sys.version_info >= (3, 12):
+        expected -= {"setuptools", "wheel"}
     assert found == expected, out

